File: AdvantageFramework.Core/Database/Views/ClientPortalAlertView.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Views
{
    public partial class ClientPortalAlertView
    {
        [Column("ALERT_ID")]
        public int AlertId { get; set; }
        [Column("SUBJECT")]
        [StringLength(254)]
        public string Subject { get; set; }
        [Column("GENERATED", TypeName = "smalldatetime")]
        public DateTime? Generated { get; set; }
        [Column("USER_NAME")]
        [StringLength(61)]
        public string UserName { get; set; }
        [Column("PRIORITY")]
        public short? Priority { get; set; }
        [Required]
        [Column("TYPE")]
        [StringLength(40)]
        public string Type { get; set; }
        [Required]
        [Column("CATEGORY")]
        [StringLength(40)]
        public string Category { get; set; }
        [Column("ALERT_TYPE_ID")]
        public int AlertTypeId { get; set; }
        [Column("ALERT_CAT_ID")]
        public int AlertCatId { get; set; }
        [Column("ATTACHMENTCOUNT")]
        public int? Attachmentcount { get; set; }
        [Column("ALERT_LEVEL")]
        [StringLength(50)]
        public string AlertLevel { get; set; }
        [Required]
        [Column("OFFICE_CODE")]
        [StringLength(4)]
        public string OfficeCode { get; set; }
        [Required]
        [Column("CL_CODE")]
        [StringLength(6)]
        public string ClCode { get; set; }
        [Required]
        [Column("DIV_CODE")]
        [StringLength(6)]
        public string DivCode { get; set; }
        [Required]
        [Column("PRD_CODE")]
        [StringLength(6)]
        public string PrdCode { get; set; }
        [Required]
        [Column("CMP_CODE")]
        [StringLength(6)]
        public string CmpCode { get; set; }
        [Column("JOB_NUMBER")]
        public int? JobNumber { get; set; }
        [Column("JOB_COMPONENT_NBR")]
        public short? JobComponentNbr { get; set; }
        [Column("SENDER_EMP_CODE")]
        [StringLength(6)]
        public string SenderEmpCode { get; set; }
        [Column("DISMISSED_DATE")]
        [StringLength(10)]
        public string DismissedDate { get; set; }
        [Column("CL_NAME")]
        [StringLength(40)]
        public string ClName { get; set; }
        [Column("DIV_NAME")]
        [StringLength(40)]
        public string DivName { get; set; }
        [Column("PRD_DESCRIPTION")]
        [StringLength(40)]
        public string PrdDescription { get; set; }
        [Required]
        [Column("CMP_NAME")]
        [StringLength(60)]
        public string CmpName { get; set; }
        [Column("BODY", TypeName = "text")]
        public string Body { get; set; }
        [Column("LOWER_SUBJECT")]
        [StringLength(254)]
        public string LowerSubject { get; set; }
        [Column("LOWER_BODY")]
        [StringLength(30)]
        public string LowerBody { get; set; }
        [Required]
        [Column("OFFICE_NAME")]
        [StringLength(30)]
        public string OfficeName { get; set; }
        [Column("JOB_DESC")]
        [StringLength(60)]
        public string JobDesc { get; set; }
        [Column("JOB_COMP_DESC")]
        [StringLength(60)]
        public string JobCompDesc { get; set; }
        [Column("PROCESSED", TypeName = "smalldatetime")]
        public DateTime? Processed { get; set; }
        [Column("CDP_CONTACT_ID")]
        public int? CdpContactId { get; set; }
        [Column("ALERT_USER")]
        [StringLength(100)]
        public string AlertUser { get; set; }
        [Column("CP_ALERT")]
        public short? CpAlert { get; set; }
        [Column("NEW_ALERT")]
        public short? NewAlert { get; set; }
        [Column("READ_ALERT")]
        public short? ReadAlert { get; set; }
    }
}

File: AdvantageFramework/Database/Views/ClientPortalAlertView.vb
Namespace Database.Views

    <Table("CP_ALERTLIST")>
    Public Class ClientPortalAlertView
        Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            AlertID
            Subject
            GeneratedDate
            UserName
            Priority
            Type
            Category
            AlertTypeID
            AlertCategoryID
            AttachmentCount
            AlertLevel
            OfficeCode
            ClientCode
            DivisionCode
            ProductCode
            CampaignCode
            JobNumber
            JobComponentNumber
            SenderEmployeeCode
            DismissedDate
            ClientName
            DivisionName
            ProductDescription
            CampaignName
            Body
            LowerSubject
            LowerBody
            OfficeName
            JobDescription
            JobComponentDescription
            Processed
            ClientContactID
            AlertUser
            ClientPortalAlert
            NewAlert
            ReadAlert

        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <Required>
        <Column("ALERT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertID() As Integer
        <MaxLength(254)>
        <Column("SUBJECT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Subject() As String
        <Column("GENERATED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GeneratedDate() As Nullable(Of Date)
        <MaxLength(61)>
        <Column("USER_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property UserName() As String
        <Column("PRIORITY")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Priority() As Nullable(Of Short)
        <Required>
        <MaxLength(40)>
        <Column("TYPE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Type() As String
        <Required>
        <MaxLength(40)>
        <Column("CATEGORY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Category() As String
        <Required>
        <Column("ALERT_TYPE_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertTypeID() As Integer
        <Required>
        <Column("ALERT_CAT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertCategoryID() As Integer
        <Column("ATTACHMENTCOUNT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AttachmentCount() As Nullable(Of Integer)
        <MaxLength(50)>
        <Column("ALERT_LEVEL", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertLevel() As String
        <Required>
        <MaxLength(4)>
        <Column("OFFICE_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property OfficeCode() As String
        <Required>
        <MaxLength(6)>
        <Column("CL_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientCode() As String
        <Required>
        <MaxLength(6)>
        <Column("DIV_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DivisionCode() As String
        <Required>
        <MaxLength(6)>
        <Column("PRD_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ProductCode() As String
        <Required>
        <MaxLength(6)>
        <Column("CMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CampaignCode() As String
        <Column("JOB_NUMBER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobNumber() As Nullable(Of Integer)
        <Column("JOB_COMPONENT_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobComponentNumber() As Nullable(Of Short)
        <MaxLength(6)>
        <Column("SENDER_EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property SenderEmployeeCode() As String
        <MaxLength(10)>
        <Column("DISMISSED_DATE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DismissedDate() As String
        <MaxLength(40)>
        <Column("CL_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientName() As String
        <MaxLength(40)>
        <Column("DIV_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DivisionName() As String
        <MaxLength(40)>
        <Column("PRD_DESCRIPTION", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ProductDescription() As String
        <Required>
        <MaxLength(128)>
        <Column("CMP_NAME", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property CampaignName() As String
		<Column("BODY")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Body() As String
        <MaxLength(254)>
        <Column("LOWER_SUBJECT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LowerSubject() As String
        <MaxLength(30)>
        <Column("LOWER_BODY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LowerBody() As String
        <Required>
        <MaxLength(30)>
        <Column("OFFICE_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property OfficeName() As String
        <MaxLength(60)>
        <Column("JOB_DESC", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobDescription() As String
        <MaxLength(60)>
        <Column("JOB_COMP_DESC", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobComponentDescription() As String
        <Column("PROCESSED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Processed() As Nullable(Of Date)
        <Column("CDP_CONTACT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientContactID() As Nullable(Of Integer)
        <MaxLength(100)>
        <Column("ALERT_USER", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertUser() As String
        <Column("CP_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientPortalAlert() As Nullable(Of Short)
        <Column("NEW_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property NewAlert() As Nullable(Of Short)
        <Column("READ_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ReadAlert() As Nullable(Of Short)


#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.AlertID.ToString

        End Function

#End Region

    End Class

End Namespace
